spec/src/main/asciidoc/architecture.asciidoc

```diff
@@ -193,47 +193,441 @@ For example, calling an `@Outgoing` annotated method from user code will not pos
 Enabling this would bypass the automatic back pressure mechanism that is one of the benefits of the specification.
 The `@Incoming` and `@Outgoing` method annotations are used to declaratively define the stream which is then run by the implementation of MicroProfile Reactive Messaging without the user's code needing to handle concerns such as subscriptions or flow control within the stream.
 
-==== Supported Bean scope
+==== Supported CDI scopes
 
+Implementations of the Reactive Messaging specification must support at least the following CDI scopes:
 
-==== Method signatures
+* `@ApplicationScoped` beans
+* `@Dependent` beans
 
-This section lists the method signatures that implementation must support.
+The following code gives an example of a bean annotated with `@ApplicationScoped`:
 
+[source,java]
+----
+@ApplicationScoped
+public class ApplicationScopeBeans {
+
+  @Outgoing("source")
+  public Publisher<Integer> source() {
+    return ReactiveStreams.of(id).buildRs();
+  }
+
+  @Incoming("source")
+  @Outgoing("output")
+  public int process(int i) {
+    return i + 1;
+  }
+
+  @Incoming("output")
+  public void sink(int v) {
+    System.out.println(v);
+  }
+}
+----
+
+Implementations can provide support for other scopes. 
+However the behavior is not defined.
+
+==== Supported method signatures
 
 The signature of message stream methods can have a number of different distinct types, offering differing levels of power and simplicity to application developers. Different shapes are supported depending on whether the method is a publisher, subscriber or processor, for example, a publishing stream supports returning MicroProfile Reactive Streams `PublisherBuilder`, but not `SubscriberBuilder`, the inverse is true for a subscribing stream.
 
+This section lists the methods signatures that must be supported by the Reactive Messaging implementation.
 Implementations must validate that the stream shape matches the `@Outgoing` and `@Incoming` annotations, if they don't, a CDI definition exception should be raised to the CDI container during initialization.
 
-===== Simple method streams
+It's important to remember that users must not call these methods directly. 
+They are invoked by the Reactive Messaging implementation following the Reactive Streams protocol.
+
+Also the method must be implemented as a non-blocking fashion.
+For blocking transformations, asynchronous variants can be used.
+
+INFO: _assembly_ time is when the Reactive Messaging implementation initializes itself and creates the different bean instances and connects them together.
+INFO: In the following lists, `Message` can be an implementation of the `Message` interface.
```

Comment:
This 'INFO' does not format to the start of a line (if that is the effect you wish).

Reply:
Missing empty line.... Fixed now.